Client/Frontend/Browser/User Scripts/ScriptFactory.swift

```diff
@@ -0,0 +1,125 @@
+// Copyright 2022 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+import WebKit
+
+/// An error representing failures in loading scripts
+enum ScriptLoadFailure: Error {
+  case notFound
+}
+
+/// A class that helps in the aid of creating and caching scripts.
+class ScriptFactory {
+  /// A shared instance to be shared throughout the app.
+  ///
+  /// - Note: Perhaps we will move this to the tab to be managed on a tab basis.
+  static let shared = ScriptFactory()
+  
+  /// Ensures that the message handlers cannot be invoked by the page scripts
+  public static let messageHandlerTokenString: String = {
+    return UUID().uuidString.replacingOccurrences(of: "-", with: "", options: .literal)
+  }()
+  
+  /// This contains cahced script sources for a script type. Avoids reading from disk.
+  private var cachedScriptSources: [ScriptSourceType: String]
+  
+  /// This contains cached altered scripts that are ready to be inected. Avoids replacing strings.
+  private var cachedDomainScriptsSources: [UserScriptType: WKUserScript]
+  
+  init() {
+    cachedScriptSources = [:]
```

Comment:
You can just do `cachedDomainScriptsSources = [UserScriptType: WKUserScript]()` instead of having to do it inside of `init`